Sources/KeyboardShortcuts/Recorder.swift

```diff
@@ -48,7 +48,16 @@ extension KeyboardShortcuts {
 		public func makeNSView(context: Context) -> NSViewType { .init(for: name, onChange: onChange) }
 
 		/// :nodoc:
-		public func updateNSView(_ nsView: NSViewType, context: Context) {}
+		public func updateNSView(_ nsView: NSViewType, context: Context) {
+			guard nsView.shortcutName != name else {
+				return
+			}
+			nsView.shortcutName = name
+			DispatchQueue.main.async {
+				// Wait for next cycle, so field be already in window
+				nsView.blur()
+			}
+		}
```

Comment:
Both these changes should be in the Cocoa view as they would be useful there too.